PORTAL/jobs/_pyperformance.py

```diff
@@ -1583,13 +1583,13 @@ def copy_to(
     ) -> "PyperfResults":
         if self._resfile is None:
             raise ValueError
-        if os.path.exists(self._resfile.filename):
-            resfile = self._resfile.copy_to(filename, resultsroot,
-                                            compressed=compressed)
+        resfile: PyperfResultsFile
+        if isinstance(filename, PyperfResultsFile):
+            resfile = filename
```

Comment:
With `_modified` set in `split_benchmarks()`:

```suggestion
        if not self._modified and os.path.exists(self._resfile.filename):
            resfile = self._resfile.copy_to(filename, resultsroot,
                                            compressed=compressed)
```

...and restore the indent on `resfile.write(self)` below.

Reply:
Also, up in `def uploadid(self):`;

`if self._resfile.uploadid:` -> `if not self._modified and self._resfile.uploadid:`